File: Core/Ecommerce.Domain/Entities/OrderItem.cs
using System.ComponentModel.DataAnnotations;
using Ecommerce.Domain.Common;

namespace Ecommerce.Domain.Entities
{
	public class OrderItem : AuditableEntity
	{
		public int Id { get; set; }
		
		public int OrderId { get; set; }
		
		[MaxLength(255)]
		public string ProductName { get; set; } = null!;
		
		[MaxLength(255)]
		public string ProductDescription { get; set; } = null!;
		
		[MaxLength(255)]
		public string ProductSku { get; set; } = null!;
		
		public int Quantity { get; set; }
		
		public double Price { get; set; }
	}
}

File: UI/ecommerce-ui/src/models/OrderItem.ts
export class OrderItem {
    id: number;
    orderId: number;
    productName: string;
    productDescription: string;
    productSku: string;
    quantity: number;
    price: number;

    constructor(
        id: number,
        orderId: number,
        productName: string,
        productDescription: string,
        productSku: string,
        quantity: number,
        price: number
    ) {
        this.id = id;
        this.orderId = orderId;
        this.productName = productName;
        this.productDescription = productDescription;
        this.productSku = productSku;
        this.quantity = quantity;
        this.price = price;
    }
}
